payu/models/cice.py

```diff
@@ -62,27 +62,63 @@ def set_model_pathnames(self):
         setup_nml = self.ice_in['setup_nml']
 
         res_path = os.path.normpath(setup_nml['restart_dir'])
+        input_dir = setup_nml.get('input_dir', None)
+
+        if input_dir is None:
+            # Default to reading and writing inputs/restarts in-place
+            input_path = res_path
+        else:
+            input_path = os.path.normpath(input_dir)
+
+        # Determine if there is a work input path from grid path
+        grid_nml = self.ice_in['grid_nml']
+        path, _ = os.path.split(grid_nml['grid_file'])
+        if path and not path == '.':
+            assert not os.path.isabs(path)
+            path = os.path.normpath(path)
+            # Get input_dir from grid_file path unless otherwise specified
+            if input_dir is None:
+                input_path = path
+
+        # Check for consistency in input paths due to cice having the same
+        # information in multiple locations
+        if not path == input_path:
+            print('Grid file path in {nmlfile} ({gridpath}) does not match '
```

Comment:
maybe prepend with `payu: error: `

Reply:
I have always wondered if we shouldn't be using some sort of logger ... but point taken. Do we have  standard way of doing this that I can refer to?

Reply:
No, there is no standard way, and yes, I wish there were.

The `payu: warning:` and `payu: error:` stuff has been slowly emergent but is by no means consistent.

Reply:
Okey doke